apps/api-journeys/src/app/modules/event/radioQuestion/radioQuestion.resolver.ts

```diff
@@ -27,12 +31,21 @@ export class RadioQuestionSubmissionEventResolver {
       input.stepId
     )
 
-    return await this.eventService.save({
-      ...input,
-      __typename: 'RadioQuestionSubmissionEvent',
-      visitorId: visitor.id,
-      createdAt: new Date().toISOString(),
-      journeyId
-    })
+    const promiseArray = [
+      this.eventService.save({
+        ...input,
+        __typename: 'RadioQuestionSubmissionEvent',
+        visitorId: visitor.id,
+        createdAt: new Date().toISOString(),
+        journeyId
+      }),
+      this.visitorService.update(visitor.id, {
+        lastRadioQuestion: input.label ?? undefined,
+        lastRadioOptionSubmission: input.value ?? undefined
+      })
+    ]
+
+    const [RadioSubmissionEvent] = await Promise.all([...promiseArray])
```

Comment:
this can be combined into a single instruction:
```suggestion
    const [radioSubmissionEvent] = await Promise.all([
      this.eventService.save({
        ...input,
        __typename: 'RadioQuestionSubmissionEvent',
        visitorId: visitor.id,
        createdAt: new Date().toISOString(),
        journeyId
      }),
      this.visitorService.update(visitor.id, {
        lastRadioQuestion: input.label ?? undefined,
        lastRadioOptionSubmission: input.value ?? undefined
      })
    ])
```